### PR TITLE
bedtools version 2.31.1

### PR DESCRIPTION
**Pull request by erinyoung**
There's a new version of bedtools, although this update seems to deal with installation issues (read more here https://github.com/arq5x/bedtools2/releases/tag/v2.31.1).

For this, I copied the Dockerfile from 2.31.0 and changed the version ARG. Everything else is the same.

Pull Request (PR) checklist:
- [X] Include a description of what is in this pull request in this message.
- [X] The dockerfile successfully builds to a test target for the user creating the PR. (i.e. `docker build --tag samtools:1.15test --target test docker-builds/samtools/1.15` )
- [X] Directory structure as name of the tool in lower case with special characters removed with a subdirectory of the version number (i.e. `spades/3.12.0/Dockerfile`)
   - [X] (optional) All test files are located in same directory as the Dockerfile (i.e. `shigatyper/2.0.1/test.sh`)
- [X] Create a simple container-specific [README.md](https://github.com/StaPH-B/docker-builds/blob/master/.github/workflow-templates/readme-template.md) in the same directory as the Dockerfile (i.e. `spades/3.12.0/README.md`)
   - [X] If this README is longer than 30 lines, there is an explanation as to why more detail was needed
- [X] Dockerfile includes the recommended [LABELS](https://github.com/StaPH-B/docker-builds/blob/master/dockerfile-template/Dockerfile#L8-L18) 
- [X] Main [README.md](https://github.com/StaPH-B/docker-builds/blob/master/README.md) has been updated to include the tool and/or version of the dockerfile(s) in this PR
- [X] [Program_Licenses.md](https://github.com/StaPH-B/docker-builds/blob/master/Program_Licenses.md) contains the tool(s) used in this PR and has been updated for any missing

### CODE REVIEW

File: bedtools/2.31.1/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+### keeping old installation cmds here in case we want to install via pre-compiled binary again in the future ###
+# RUN cd /usr/local/bin && \
+#     wget https://github.com/arq5x/bedtools2/releases/download/v${BEDTOOLS_VER}/bedtools.static && \
+#     mv bedtools.static bedtools && \
+#     chmod +x bedtools && \
+#     mkdir /data
+
+### test stage ###
+FROM builder as test
```
Comment (Kincekara) [inline bedtools/2.31.1/Dockerfile:52]: "FROM app as test"? 
We need to test the app to see any missing libraries in the app stage. So, the test stage needs to be after the app stage.

File: bedtools/2.31.1/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+# test scripts expect to be run from the bedtools root dir
+WORKDIR /bedtools2-${BEDTOOLS_VER}
+
+# commenting out ulimit command in test/test.sh (gitpod complains)
+# run tests included with bedtools code
+RUN sed -i 's/ulimit/#ulimit/g' test/test.sh && \
+make test
+
+### final app stage ###
```
Comment (Kincekara) [inline bedtools/2.31.1/Dockerfile:65]: This part needs to be done before the test stage to be used in the test stage.

**Comment by Kincekara** [comment]
I believe those mistakes come from the previous image. Although it can work this way, fixing it in this PR would be better.

File: bedtools/2.31.1/Dockerfile (hunk 1 above)
Comment (Kincekara) [reply bedtools/2.31.1/Dockerfile:52]: You will need to download the test files again in the test stage.

Comment (erinyoung) [reply bedtools/2.31.1/Dockerfile:52]: bedtools has a lot of subcommands, but the internal tests of bedtools has a dependency from builder.

I think when we did this one before, we tested it with some files after.

Let me get this one up to standard practice, though.

File: bedtools/2.31.1/Dockerfile (hunk 2 above)
Comment (erinyoung) [reply bedtools/2.31.1/Dockerfile:48]: I've decided to move this into the builder stage and then I added a test to the test stage

**Comment by erinyoung** [comment]
@Kincekara , Thank you for your suggestions!

I have made some adjustments by combined the old test stage into the builder stage - because I do like how extensively it gets tested.

I also moved the labels into the app stage and created a test stage built from app. 

Let me know if I should make any other changes!

File: bedtools/2.31.1/Dockerfile (hunk 2 above)
Comment (Kincekara) [reply bedtools/2.31.1/Dockerfile:48]: That makes sense!

**Comment by Kincekara** [comment]
@erinyoung Thank you for all your adjustments! Moving the internal test to the builder stage and creating a separate test is a great idea. Overall, the image looks fantastic.

**Review by Kincekara — APPROVED** [review]

**Comment by Kincekara** [comment]
Thank you for your PR! 
You can check the deploy status from here: https://github.com/StaPH-B/docker-builds/actions/runs/6961689944

**Pull request merged**